python/pyspark/sql/tests/pandas/test_pandas_udf_scalar.py

```diff
@@ -1320,6 +1322,7 @@ def f4_iter(it):
             self.assertEqual(expected_multi, df_multi_1.collect())
             self.assertEqual(expected_multi, df_multi_2.collect())
 
+    @unittest.skipIf(not have_grpcio, grpcio_requirement_message)
```

Comment:
@xinrong-meng and @zhengruifeng this is from https://github.com/apache/spark/commit/7c7b9585a2aba7bbd52c197b07ed0181ae049c75. Can we separate the connect test into separate connect directory? Ideally this place should not require connect dependencies

Reply:
+1

Reply:
Sounds good I'll adjust that.

Reply:
Adjusted in https://github.com/apache/spark/pull/44886.

Reply:
According to the test results, regarding the error of not installing `grpcio`, we have fixed it. 
Thanks, @xinrong-meng.

Reply:
Thank you @panbingkun !